src/sidebar/cross-origin-rpc.js

```diff
@@ -93,8 +107,34 @@ function start(store, settings, $window) {
   }
 }
 
+/**
+ * Queues all incoming RPC requests so they can be handled later.
+ *
+ * @param {object} event - RPC event payload
+ */
+function preStartMessageListener(event) {
+  preStartQueue.push(event);
+}
+
+/**
+ * Start listening to incoming RCP requests right away so they don't timeout. These
+ * requests are saved until the server starts up and then they can be handled accordingly.
+ *
+ * Why?
+ *
+ * This allows the client to fetch its config from the parent app frame and potentially
+ * receive new unsolicited incoming requests that the parent app may send before this
+ * RCP server starts.
+ *
+ * @param {Window} $window - Test seam
+ */
+function preStart($window = window) {
```

Comment:
`$window` is a built-in feature in Angular, but outside of Angular code the preferred convention for this kind of test seam (for DOM objects like `window` or `document`) is to add an underscore after the name (eg. `window_ = window`).

Reply:
Thanks, was wondering about that